pkg/service/storage/accessor.go

```diff
@@ -0,0 +1,36 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aerospike/aerospike-backup-service/v2/pkg/model"
+	"github.com/aerospike/backup-go"
+)
+
+// Accessor interface abstracts storage layer.
+type Accessor interface {
+	supports(storage model.Storage) bool
+	createReader(ctx context.Context, storage model.Storage, path string, isFile bool, filter Validator,
+	) (backup.StreamingReader, error)
+	createWriter(ctx context.Context, storage model.Storage, path string, isFile, isRemoveFiles, withNested bool,
+	) (backup.Writer, error)
+}
+
+var accessors []Accessor
```

Comment:
I believe you have considered another option than a global variable.

Reply:
yes, I just though it's easier